src/feature/nodelist/networkstatus.c

```diff
@@ -1750,6 +1742,41 @@ networkstatus_set_current_consensus_from_ns(networkstatus_t *c,
 }
 #endif /* defined(TOR_UNIT_TESTS) */
 
+/**
+ * Helper: Read a the current consensus of type <b>flavor</b> from
+ * <b>fname</b>.  Flags and return values are as for
+ * networkstatus_set_current_consensus().
+ **/
+static int
+reload_consensus_from_file(const char *fname,
+                           const char *flavor,
+                           unsigned flags,
+                           const char *source_dir)
+{
+  tor_mmap_t *map = tor_mmap_file(fname);
+  if (!map)
+    return 0;
+
+  int rv = networkstatus_set_current_consensus(map->data, map->size,
+                                               flavor, flags, source_dir);
+#ifdef _WIN32
+  if (rv < 0 && tor_memstr(map->data, map->size, "\r\n")) {
```

Comment:
We need to do this check *before* we parse the consensus, otherwise users will see a warning from networkstatus_set_current_consensus().

Reply:
In particular, I think we should either call:
```
networkstatus_set_current_consensus(map->data, ...
```
or
```
char *content = read_file_to_str(fname, RFTS_IGNORE_MISSING, NULL);
rv = networkstatus_set_current_consensus(content, ...
```
But not both.